de.variantsync.core/src/main/java/de/variantsync/core/ast/LineBasedParser.java

```diff
@@ -0,0 +1,51 @@
+package de.variantsync.core.ast;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class LineBasedParser {
+
+	public LineBasedParser() {}
+
+	public AST<LineGrammar, String> parseDirectory(Path folder) throws IOException {
+		AST<LineGrammar, String> result;
+		if (Files.isDirectory(folder)) {
+			result = new AST<>(LineGrammar.Directory, folder.getFileName().toString());
```

Comment:
Is `toString()` necessary? If yes, why don't call with i.e. Path object

Reply:
Yes it is. The return type is `AST<LineGrammar, String>` which means all nodes' values will be of type String. Storing paths or files instead of strings is a good idea for nodes of type `LineGrammar.Directory` or `LineGrammar.File`, though. This would require exceptional behaviour for certain node types. I don't know how to implement that generically and transparently.